cluster/matrix.py

```diff
@@ -123,10 +159,17 @@ def genmatrix(self, num_processes=1):
                         num_tasks_completed += 1
                 else:
                     # Otherwise do it here, in line
+                    """
```

Comment:
I only noticed this after merging. Can you remove this string literal please? It may also be useful to quickly run a PEP8 checker over your changes and fix the biggest issues.

The code works and is merged, so there's no hurry. It just keeps the code a bit cleaner.